src/main/java/ljv/ObjSettings.java

```diff
@@ -0,0 +1,109 @@
+                //- accessible, we want an IllegalAccessException to be raised
+                //- (and caught).  It is not correct to return true if
+                //- field.getType( ).isPrimitive( )
+                Object val = field.get(obj);
+                if (field.getType().isPrimitive() || canTreatAsPrimitive(ljv, val))
+                    //- Just calling ljv.canTreatAsPrimitive is not adequate --
+                    //- val will be wrapped as a Boolean or Character, etc. if we
+                    //- are dealing with a truly primitive type.
+                    return true;
+            } catch (IllegalAccessException e) {
+                e.printStackTrace();
+            }
+        }
+
+        return false;
+    }
+
+    public boolean hasPrimitiveFields(Field[] fs, Object obj) {
+        for (Field f : fs)
+            if (fieldExistsAndIsPrimitive(f, obj))
+                return true;
+        return false;
+    }
+
+    private static boolean redefinesToString(Object obj) {
+        Method[] ms = obj.getClass().getMethods();
+        for (Method m : ms)
+            if (m.getName().equals("toString") && m.getDeclaringClass() != Object.class)
+                return true;
+        return false;
+    }
+
+
+    public String className(Object obj, boolean useToStringAsClassName) {
+        if (obj == null)
+            return "";
+
+        Class<?> c = obj.getClass();
+        if (useToStringAsClassName && redefinesToString(obj))
+            return Quote.quote(obj.toString());
+        else {
+            String name = c.getName();
+            if (!ljv.isShowPackageNamesInClasses() || c.getPackage() == LJV.class.getPackage()) {
+                //- Strip away everything before the last .
+                name = name.substring(name.lastIndexOf('.') + 1);
+
+                if (!ljv.isQualifyNestedClassNames())
+                    name = name.substring(name.lastIndexOf('$') + 1);
+            }
+            return name;
+        }
+    }
+
+    public boolean canTreatAsPrimitive(LJV LJV, Object obj) {
+        return obj == null || canTreatClassAsPrimitive(LJV, obj.getClass());
+    }
+
+
+    public boolean canTreatClassAsPrimitive(LJV LJV, Class<?> cz) {
+        if (cz == null || cz.isPrimitive())
+            return true;
+
+        if (cz.isArray())
+            return false;
+
+        do {
+            if (LJV.isTreatsAsPrimitive(cz)
+                    || LJV.isTreatsAsPrimitive(cz.getPackage())
+            )
+                return true;
+
+            if (cz == Object.class)
+                return false;
+
+            Class<?>[] ifs = cz.getInterfaces();
+            for (Class<?> anIf : ifs)
+                if (canTreatClassAsPrimitive(LJV, anIf))
+                    return true;
+
+            cz = cz.getSuperclass();
+        } while (cz != null);
+        return false;
+    }
+
+    public boolean looksLikePrimitiveArray(Object obj, LJV LJV) {
```

Comment:
You don't need `LJV` as an argument here, it's already available as a class member